extensions/dust-tt/src/pickWorkspace.tsx

```diff
@@ -2,102 +2,165 @@ import {
 import { getRandomGreetingForName, getWorkspaceId } from "./utils";
 import { useCallback } from "react";
+import { DustAPI } from "@dust-tt/client";
+import { withDustClient, getCurrentToken } from "./dust_api/oauth";
+
+const DUST_US_URL = "https://dust.tt";
+const DUST_EU_URL = "https://eu.dust.tt";
+
+type WorkspaceWithRegion = {
+  sId: string;
+  name: string;
+  role: string;
+  region: string;
+};
 
 export default withDustClient(function PickWorkspaceCommand() {
   const navigation = useNavigation();
-  const dustAPI = getDustClient();
-  const preferences = getPreferenceValues<ExtensionPreferences>();
-  const isUsingOauth = preferences.connexionFlow === "oauth";
 
-  const { data: user, isLoading: isLoadingUser } = usePromise(
-    async (isUsingOauth) => {
-      if (isUsingOauth) {
-        const r = await dustAPI.me();
-        if (r.isErr()) {
-          showFailureToast(`Could not get user information: ${r.error.message}`);
-          return undefined;
-        } else {
-          // Keep the user updated in the local storage as much as possible
-          await LocalStorage.setItem("user", JSON.stringify(r.value));
-          return r.value;
-        }
-      } else {
-        return undefined;
+  const { data: workspacesData, isLoading: isLoadingWorkspaces } = usePromise(async () => {
+    const usWorkspaces: WorkspaceWithRegion[] = [];
+    const euWorkspaces: WorkspaceWithRegion[] = [];
+    let user = null;
+    const token = getCurrentToken();
+
+    // Fetch from US region
+    try {
+      const usAPI = new DustAPI({ url: DUST_US_URL }, { apiKey: token, workspaceId: "" }, console);
+      const usResult = await usAPI.me();
+      if (usResult.isOk()) {
+        usWorkspaces.push(
+          ...usResult.value.workspaces.map((ws) => ({
+            ...ws,
+            region: "us-central1",
+          })),
+        );
+        user = usResult.value;
       }
-    },
-    [isUsingOauth],
-  );
+    } catch (error) {
+      // The user might not be exist in US region, so we can ignore errors here.
+    }
 
-  const { data: workspaceId, isLoading: isLoadingWorkspace } = usePromise(
-    async (isUsingOauth, isLoadingUser) => {
-      if (isUsingOauth && !isLoadingUser) {
-        return getWorkspaceId();
+    // Fetch from EU region
+    try {
+      const euAPI = new DustAPI({ url: DUST_EU_URL }, { apiKey: token, workspaceId: "" }, console);
+      const euResult = await euAPI.me();
+      if (euResult.isOk()) {
+        euWorkspaces.push(
+          ...euResult.value.workspaces.map((ws) => ({
+            ...ws,
+            region: "europe-west1",
+          })),
+        );
+        // Use EU user info if we didn't get US info
+        if (!user) {
+          user = euResult.value;
+        }
       }
-      return undefined;
-    },
-    [isUsingOauth, isLoadingUser],
-  );
+    } catch (error) {
+      // The user might not be exist in EU region, so we can ignore errors here.
+    }
```

Comment:
Note that in one single call to any region to /me, you also get the list of all organizations , from both EU and US - with workspaceId as externalId and region in metadata. And you'll also get the "selectedWorkspace" which is the one taht was selected in org chooser